docs/data_cycling.md

```diff
@@ -0,0 +1,51 @@
+# Data retention policies
+
+## On Perfherder
+
+On a daily basis, Perfherder expires data out of several reasons:
+
+* data provides lower & lower interest as it grows older
+* data accumulates very fast (>1 million new data points are ingested daily)
+* query latency degrades in time, down to intermittent timeouts
```

Comment:
Not sure I understand "query latency degrades in time, down to intermittent timeouts". Does this mean "query latency degrades with time as stored data increases, causing intermittent timeouts"???

Reply:
I now realize this comment is a bit more worrying than intended.
Our apps' queries (from web APIs, background workers etc) have been stable even for bigger amounts of perf data (more than x5 times we currently have).

But here I'm mostly referring to random manual queries done via ReDash (the business intelligence like ones). These tend to be complex & inspect data in ways that are not (yet) optimized (& likely will never be). For example: indexes that are good for BI queries aren't always useful for the app' s performance.

I think I'll simply remove the `down to intermittent timeouts` part.